src/transports/smart_protocol.c

```diff
@@ -285,6 +285,9 @@ static int fetch_setup_walk(git_revwalk **out, git_repository *repo)
 	git_revwalk_sorting(walk, GIT_SORT_TIME);
 
 	for (i = 0; i < refs.count; ++i) {
+                const git_oid *oid;
+                git_object *obj, *oobj;
```

Comment:
These object are leaking as soon as we encounter an error. You could instead just initialize them to `NULL`, `git_object_free` at the beginning of each iteration as well as freeing in the `on_error` section. This requires us to define these variables at the beginning of the function.